laws/src/main/scala/cats/laws/discipline/DecidableTests.scala

```diff
@@ -46,12 +46,18 @@ trait DecidableTests[F[_]] extends ContravariantMonoidalTests[F] {
     iso: SemigroupalTests.Isomorphisms[F]
   ): RuleSet =
     new RuleSet {
+      implicit val ArbNothing: Arbitrary[Nothing => Nothing] = Arbitrary(Gen.const((n: Nothing) => n))
+      implicit val eqNothing: Eq[F[Nothing]] = Eq.by[F[Nothing], Unit]((_: F[Nothing]) => ())
```

Comment:
Ha these are rigged 😂 actually I think `Eq[F[Nothing]]` needs be an implicit constraint of the tests, because we don't know how equality of `F[_]` is supposed to work. Then, maybe we can add `Eq[Nothing]` to Cats?

Reply:
Sounds good hah. I went this route mainly because scoping these sort of bizarre instances tightly seems least risky, but yeah I see your point on the constraint having to be user accessible for laws testing new instances.

Reply:
I do want to call out - just looking at this finally today, this is going to add some pretty weird things if we want to run these tests against the cats instances, e.g. `Eq[Eq[Nothing]]` is becomes a requirement... These feel like they should be test scoped, but I don't know if we have a standard way of doing that? Maybe you can advice @armanbilge

Reply:
`Eq[Eq[Nothing]]` seems fine (should always return `true` I reckon), although I agree it should be scoped. I think it's fine to put it directly in the test suite for now. I can't remember where these things live, I'm sure we have some instances for `Invariant` tests or something.